mm2src/coins/z_coin.rs

```diff
@@ -678,7 +678,7 @@ impl ZCoin {
         else {
             return Ok(false);
         };
-        if &address == expected_address {
+        if &address != expected_address {
```

Comment:
So this was the bug and we had to write ~300 lines of test code to cover it ? :)